src/Sarif.Viewer.VisualStudio.Core/Telemetry/KeyEventTelemetry.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+using Microsoft.Sarif.Viewer.Shell;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.Sarif.Viewer.Telemetry
+{
+    internal class KeyEventTelemetry
+    {
+        internal const string Product = "VS/Extensions/SarifViewer/KeyEvents/";
+
+        private static string vsVersion;
+        private static string extensionVersion;
+
+        private readonly ITelemetryClient telemetryClient;
+
+        public static string VsVersion
+        {
+            get
+            {
+                if (SarifViewerPackage.IsUnitTesting)
+                {
+                    return string.Empty;
+                }
+
+                vsVersion ??= VsUtilities.GetVsVersion() ?? string.Empty;
+                return vsVersion;
+            }
+        }
+
+        public static string ExtensionVersion
```

Comment:
>ExtensionVersion [](http://example.com/codeflow?start=29&length=16)

Is it possible to use the actual extension version here? Might be easier to associate with events

Reply:
as suggested, read the actual version from extension manifest file